# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/FinalizationObservers.h
# Commit: 59ae32a4a0a3
# Full Hash: 59ae32a4a0a3be6160a757099e38bbc5f52db4b1
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-01 16:17:59
# Regressor Bug: 1755725
# File Overlap Count: 1
# Description:
#   Bug 1755725 - Part 1: Factor out crossZoneWrappers checks and do them before we calculate sweep groups r=sfink
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D139835
# ==============================================================================

diff -r ae499200e145 -r 59ae32a4a0a3 js/src/gc/FinalizationObservers.h
--- a/js/src/gc/FinalizationObservers.h	Tue Mar 01 08:27:54 2022 +0000
+++ b/js/src/gc/FinalizationObservers.h	Tue Mar 01 08:28:41 2022 +0000
@@ -93,6 +93,10 @@
   void traceRoots(JSTracer* trc);
   void traceWeakEdges(JSTracer* trc);
 
+#ifdef DEBUG
+  void checkTables() const;
+#endif
+
  private:
   bool addCrossZoneWrapper(JSObject* wrapper);
   void removeCrossZoneWrapper(JSObject* wrapper);